### DISCUSSION TITLE
Inspecting dagster results

### DISCUSSION THREAD

**Discussion by eguiraud** [dagster-io/dagster#21112] [Q&A]
Hello, I'm evaluating dagster for use at $work. I built a little example dag with the ops and jobs API, and ran the DAG from the UI.

According to the UI, the result gets saved at `$DAGSTER_HOME/storage/53539ba7-691c-469b-a48c-909eb4dd222a/mytask/result`. Now I would like to inspect that result, or anyways load it back for further processing, independently of dagster. Is that possible? is the binary format of `result` documented somewhere?

Thank you in advance!

EDIT:
maybe a better way to pose the question: what's the simplest way to get back from dagster that result to apply further processing, ideally outside of dagster?

**Comment by yuhan** [comment]
A few options:
1. For the simplest way, given you might be using the default filesystem I/O manager which just pickles the outputs, you could just "reverse eng" it:
```python
import pickle

# Replace this with the actual file path to your result
file_path = '$DAGSTER_HOME/storage/53539ba7-691c-469b-a48c-909eb4dd222a/mytask/result'

with open(file_path, 'rb') as file:
    result = pickle.load(file)
```
2. If you want to be more robust and scalable, and prefer a more human-readable format like JSON or CSV, you could implement a custom IO manager ([example](https://docs.dagster.io/concepts/io-management/io-managers#a-custom-io-manager-that-stores-pandas-dataframes-in-tables)) that serializes the data in your preferred format. You would then configure your job to use this custom IO manager instead of the default one:
  - Implement a custom IO manager that stores the data in a format that is suitable for your needs (e.g., JSON, CSV).
  - Configure your job to use this custom IO manager.
  - Access the stored data directly based on your customization.
